internal/webapp/vis/src/internal/D3Renderer.js

```diff
@@ -2926,6 +2926,146 @@ LABKEY.vis.internal.D3Renderer = function(plot) {
         }
     };
 
+    var renderStackedBarPlotGeom = function(data, geom) {
+        var layer = getLayer.call(this, geom), barWrappers, binWidth, barWidth, offsetWidth,
+                xCategories, xSubCategories, numXSubCategories, rects, hoverFn, heightFn,
+                xAcc, colorAcc, yAcc, yZero;
+
+        if (!geom.xSubScale || !geom.xSubAes) {
+            console.error('Stacked Bar Plots require xSubScale and xSubAes.');
+            return;
+        }
+
+        if (geom.xScale.scaleType !== scaleType.discrete || geom.xSubScale.scaleType !== scaleType.discrete) {
+            console.error('Stacked Bar Plots not supported for continuous data yet.');
+            return;
+        }
+
+        xCategories = geom.xScale.scale.domain();
+        xSubCategories = geom.xSubScale.scale.domain();
+        numXSubCategories = xSubCategories.length;
+
+        var pivotData = {};
+        data.forEach(function(d) {
+            if (!pivotData[d[geom.xSubAes.value]]) pivotData[d[geom.xSubAes.value]] = {};
+            pivotData[d[geom.xSubAes.value]][d[geom.xAes.value]] = d.value;
+        });
+        var pivotDataArr = [];
+        Object.keys(pivotData).forEach(function(key) {
+            var dataObj = pivotData[key];
+            dataObj[geom.xAes.value] = key;
+            pivotDataArr.push(dataObj);
+        });
+        var stackData = d3.layout.stack()(xCategories.map(function(category) {
+            return pivotDataArr.map(function(d) {
+                var dataObj = {};
+                dataObj[geom.xAes.value] = category;
+                dataObj[geom.xSubAes.value] = d[geom.xAes.value];
+
+                var value = d[category] ? d[category] : 0;
+                dataObj['value'] = value;
+                dataObj['y'] = +value;
+
+                return dataObj;
+            });
+        }));
+        data = stackData.reduce(function(prev, current) {
+            // keep track of which data points are from the last set, for the showValues case below
+            for (var index in prev) {
+                prev[index].showValue = false;
+            }
+            for (var index in current) {
+                current[index].showValue = true;
+            }
+            return prev.concat(current);
+        }, []);

```

Comment:
Instead of iterating through all of the previous values on every pass, couldn't we just detect when we're on the last set, and then set the values?
```suggestion
        data = stackData.reduce(function(prev, current, index) {
            // Only showValue for the last set
            if (index === stackData.length - 1) {
                prev.forEach(function (item) {
                    item.showValue = false;
                });
    
                current.forEach(function (item) {
                   item.showValue = true;
                });
            }
            return prev.concat(current);
        }, []);

```

Reply:
yeah that might work well. I'll give it a try

Reply:
done